src/curve/stable/StablePair.sol

```diff
@@ -200,7 +186,9 @@ contract StablePair is UniswapV2ERC20, ReentrancyGuard, IAssetManagedPair {
     /// @dev Mints LP tokens - should be called via the router after transferring tokens.
     /// The router must ensure that sufficient LP tokens are minted by using the return value.
     function mint(address to) public nonReentrant returns (uint256 liquidity) {
-        (uint256 _reserve0, uint256 _reserve1, ) = _getReserves();
+        _syncManaged();
+
+        (uint112 _reserve0, uint112 _reserve1, ) = _getReserves();
```

Comment:
Was this a bug or why did the ordering change (maybe just for cleanness)?

Reply:
so early today when I was doing the refactoring I realized that I did not call `_syncManaged` for `mint` and `burn`, hence the addition

Reply:
In general, I'm a big believer in the approach:

1. Find bug
2. Write unit test that fails, proving the bug
3. Fix the bug
4. If the unit test now passes you've proven the fix and the bug

This is key to quality & security over time, the alternative is hasty patches and vulnerability over time. I implemented this approach religiously in `vault` and it saved me from some silly regression bugs a few times.

Reply:
Yes I agree 100%. Apologies, was a lil overwhelmed with all the moving parts. 

Enhanced one of the test cases to catch a wrong/missing implementation.